src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -750,6 +750,17 @@ private void InitializeCustomNodeManager()
                 if (customNodeSearchRegistry.Contains(info.FunctionId))
                     return;
 
+                var elements = SearchModel.SearchEntries.OfType<CustomNodeSearchElement>().Where(
+                                x =>
+                                {
+                                    return string.Compare(x.Path, info.Path, StringComparison.OrdinalIgnoreCase) == 0;
+                                }).ToList();
+                foreach (var element in elements)
+                {
+                    SearchModel.Remove(element);
```

Comment:
@aosyatnik 
Great idea! An update operation should be faster than a remove operation and an add operation.